x/evm/abci.go

```diff
@@ -542,13 +543,103 @@ func handleConfirmedEvents(ctx sdk.Context, bk types.BaseKeeper, n types.Nexus,
 	}
 }
 
+func validateGeneralMessage(ctx sdk.Context, ck types.ChainKeeper, n types.Nexus, m types.MultisigKeeper, chain nexus.Chain, msg nexus.GeneralMessage) (sdk.Int, multisig.KeyID, error) {
+	chainID := funcs.MustOk(ck.GetChainID(ctx))
+
+	keyID, ok := m.GetCurrentKeyID(ctx, chain.Name)
+	if !ok {
+		return chainID, keyID, fmt.Errorf("current key not set")
+	}
+
+	if !n.IsChainActivated(ctx, chain) {
+		return chainID, keyID, fmt.Errorf("destination chain de-activated")
+	}
+
+	if len(msg.Receiver) != 0 && !common.IsHexAddress(msg.Receiver) {
```

Comment:
```suggestion
	if !common.IsHexAddress(msg.Receiver) {
```
0 length check isn't needed